### PR TITLE
Add client fields as local constants on assessments

### PR DESCRIPTION
**Pull request by martha**
## Description

PT issue https://www.pivotaltracker.com/story/show/187136045
This change allows us to pull in info about the client -- DOB, name, and SSN -- to autofill in assessments.

I have a few questions:
- Is pulling SSN in here like this okay? The client is from the enrollment, so the most obvious way to do it (and what I've implemented here) is to add client's ssn on the enrollment fragment. But of course I can think of lots of reasons not to do that, and it's a huge change! Is there a better way?
- Also related to SSN, should it be auto-filled but hidden unless the user clicks, like on the client profile?
- And, should we do any checking of the `canViewFullSsn` or `canViewPartialSsn` before passing the ssn around?
- Just to make sure, it looks like the UHA form has space for a lot of household members, but this change is only putting the info for the head of household into local constants. Is this okay?

## Type of change
[//]: # 'remove options that are not relevant'
- [ ] Bug fix
- [x] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/modules/assessments/components/AssessmentForm.tsx [hunk 1]
```diff
@@ -147,9 +148,14 @@ const AssessmentForm: React.FC<Props> = ({
       entryDate: enrollment.entryDate,
       exitDate: enrollment.exitDate,
       projectName: enrollment.project.projectName,
+      clientFirstName: client.firstName,
```
Comment (martha) [inline src/modules/assessments/components/AssessmentForm.tsx:151]: I just realized that only ssn and dob were requested in the ticket, should I remove the name fields?

File: src/api/operations/client.fragments.graphql [hunk 2]
```diff
@@ -80,6 +80,12 @@ fragment ClientNameDobVet on Client {
   veteranStatus
 }
 
+fragment ClientNameDobSsn on Client {
```
Comment (gigxz) [inline src/api/operations/client.fragments.graphql]: Interesting, we don't usually take this approach of defining a fragment that is not used in any queries. I think we should either (1) define this type in the code, ` type ClientNameDobSsn = ClientNameFragment & {ssn?: string, dob?:string}` or, (2)  add `... ClientNameDobSsn` fragment to the AllEnrollmentDetails.client fields

File: src/modules/assessments/components/AssessmentForm.tsx (hunk 1 above)
Comment (gigxz) [reply src/modules/assessments/components/AssessmentForm.tsx:151]: We can keep them, thanks

**Comment by gigxz** [comment]
> Is pulling SSN in here like this okay? The client is from the enrollment, so the most obvious way to do it (and what I've implemented here) is to add client's ssn on the enrollment fragment. But of course I can think of lots of reasons not to do that, and it's a huge change! Is there a better way?

I think that is appropriate

> Also related to SSN, should it be auto-filled but hidden unless the user clicks, like on the client profile?

That would be nice to have but out of scope I think

> And, should we do any checking of the canViewFullSsn or canViewPartialSsn before passing the ssn around?

No need, the ssn value will be null if user lacks access, or partial if thats what the user can see

> Just to make sure, it looks like the UHA form has space for a lot of household members, but this change is only putting the info for the head of household into local constants. Is this okay?

Yep

**Review by gigxz — APPROVED** [review]

**Pull request merged**